### PR TITLE
Improve types and stop reading selections from crashing

### PR DESCRIPTION
**Pull request by vincentmvdm**
## Changes

* Adds missing types, fixes incorrect types, and expands `readBlackList` based on the latest Figma plugin types
    * I made the changes by first diffing Figma's plugin types `v49` and `v58` with each other, and then diffing Figma's plugin types `v58` with our types
* Stops **reading** selections from crashing. This does **not** fix insertion
    * I've started on the insertion fixes but want to ship this first to unblock features
* Improves UI styling so that copying the displayed JSON doesn't copy unwanted things, like the insert button's text
* Removes code that saves recently inserted text to clientStorage because most selections are too big (=crash)

## Test plan

* Verified that there are no TS errors and successfully ran the existing tests
* Went into multiple files and tested the plugin on complex selections with components, instances, etc.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -6,27 +6,34 @@ import updateImageHashes from "./updateImageHashes";
...
   "hasMissingFont",
   "absoluteTransform",
   "absoluteRenderBounds",
-  "vectorNetwork"
+  "absoluteBoundingBox",
+  "vectorNetwork",
+  // Figma exposes this but plugin types don't support it yet
+  "playbackSettings",
+  // Deprecated but Figma still exposes it
+  "horizontalPadding"
```
Comment (darknoon) [inline src/index.ts:29]: lgtm

File: src/updateImageHashes.test.ts [hunk 2]
```diff
@@ -1,83 +1,126 @@
...
+  strokeWeight: 1,
+  strokeAlign: "INSIDE",
+  strokeJoin: "MITER",
   dashPattern: [],
+  strokeCap: "NONE",
+  strokeMiterLimit: 4,
+  fillGeometry: [
+    {
+      windingRule: "NONZERO",
+      data: "M0 0L375 0L375 812L0 812L0 0Z"
```
Comment (darknoon) [inline src/updateImageHashes.test.ts:62]: this is based on running the plugin right?

**Review by darknoon — APPROVED** [review]

File: src/updateImageHashes.test.ts (hunk 2 above)
Comment (vincentmvdm) [reply src/updateImageHashes.test.ts:62]: Yes :) Just making sure that this test matches reality and e.g. doesn't include keys we don't allow.

File: src/figma-json.ts [hunk 3]
```diff
@@ -536,22 +677,29 @@ export type HandleMirroring = "NONE" | "ANGLE" | "ANGLE_AND_LENGTH";
 export interface MinimalStrokesMixin {
   strokes: ReadonlyArray<Paint>;
   strokeStyleId: string;
-  strokeWeight: number;
+  strokeWeight: number | Mixed;
```
[the remark this answers is not in the document]
Comment (vincentmvdm) [reply src/figma-json.ts:680]: Yeah I hear you. Current implementation is the result of simply swapping Figma's Mixed type for our own Mixed type, like do we everywhere else.

**Pull request merged**